### PR TITLE
Set data type for `NULL` params

### PR DESCRIPTION
**Pull request by esaounkine**
In relation to issue #14.

The tests will always succeed with H2, but when executed with Postgres, the test `nullParamsJdbcHandling` should testify that the issue is closed.

When/If this PR is merged, please create a new release.

### CODE REVIEW

File: src/main/kotlin/kotliquery/Session.kt [hunk 1]
```diff
@@ -29,7 +30,7 @@ open class Session(
 
     private fun PreparedStatement.setParam(idx: Int, v: Any?) {
         if (v == null) {
-            this.setObject(idx, null)
+            this.setNull(idx, Types.VARCHAR)
```
Comment (seratch) [inline src/main/kotlin/kotliquery/Session.kt]: If I understand correctly, this call doesn't work for non-string columns.

Comment (esaounkine) [reply src/main/kotlin/kotliquery/Session.kt]: Postgres only relies on this information when the type of param cannot be figured out differently. As a matter of fact, there's no other non-intrusive way to figure out the supposed type of current `null` value.

Comment (esaounkine) [reply src/main/kotlin/kotliquery/Session.kt]: Maybe just my case-matching didn't work out.

**Comment by seratch** [comment]
Thanks, your changes look reasonable. Also, introducing the typed params looks so cool 👍

**Pull request merged**